File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/aws/ssm/ParameterKeyLookup.java
package uk.nhs.nhsx.core.aws.ssm;

import uk.nhs.nhsx.core.signature.KeyId;

public class ParameterKeyLookup implements KeyLookup {

    private final Parameter<KeyId> parameter;

    public ParameterKeyLookup(Parameters parameters, ParameterName parameterName) {
        this.parameter = parameters.parameter(parameterName, KeyId::of);
    }

    public KeyId getKmsKeyId() {
        return parameter.value();
    }
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/aws/ssm/ParameterKeyLookup.kt
package uk.nhs.nhsx.core.aws.ssm

import uk.nhs.nhsx.core.signature.KeyId

class ParameterKeyLookup(
    parameters: Parameters,
    parameterName: ParameterName
) : KeyLookup {
    private val parameter = parameters.parameter(parameterName, KeyId::of)

    override fun kmsKeyId(): KeyId = parameter.value()
}
